xbmc/video/dialogs/GUIDialogVideoBookmarks.cpp

```diff
@@ -380,15 +378,37 @@ bool CGUIDialogVideoBookmarks::AddBookmark(CVideoInfoTag* tag)
   bookmark.player = g_application.GetCurrentPlayer();
 
   // create the thumbnail image
-  float aspectRatio = appPlayer->GetRenderAspectRatio();
-  int width = BOOKMARK_THUMB_WIDTH;
-  int height = (int)(BOOKMARK_THUMB_WIDTH / aspectRatio);
-  if (height > (int)BOOKMARK_THUMB_WIDTH)
+  const float aspectRatio{appPlayer->GetRenderAspectRatio()};
+  CRect srcRect{}, renderRect{}, viewRect{};
+  appPlayer->GetRects(srcRect, renderRect, viewRect);
+  const unsigned int srcWidth{static_cast<unsigned int>(srcRect.Width())};
+  const unsigned int srcHeight{static_cast<unsigned int>(srcRect.Height())};
+  const unsigned int renderWidth{static_cast<unsigned int>(renderRect.Width())};
+  const unsigned int renderHeight{static_cast<unsigned int>(renderRect.Height())};
+  const unsigned int viewWidth{static_cast<unsigned int>(viewRect.Width())};
+  const unsigned int viewHeight{static_cast<unsigned int>(viewRect.Height())};
+
+  if (!srcWidth || !srcHeight || !renderWidth || !renderHeight || !viewWidth || !viewHeight)
+    return false;
+
+  // FIXME: the renderer sets the scissors to the size of the screen (provided by graphiccontext),
```

Comment:
I don't think that the scissor is the limiting factor. I've skimmed through the code for the capture of a frame in GL. From what I can tell, we render to the default framebuffer, which is the limiting element. Scissors are just set to the FB size.

Even though we could render to a separate buffer (which we should if we want to keep the route), I think the current way is not optimal. Ideally, we would capture the original planes, and not the mangled output which might introduce scaling artifacts (amongst other problems). But that would require re-engineering a lot of the capture code.

tl;dr: scissors are not the issue, as far as I see.

Reply:
Scissors are the limiting factor in Windows. A dedicated texture of the requested thumb size is allocated (ie could be 4k even if screen is 1080p) for render. The rendering is then clipped by the scissors to screen size, resulting in the reported issue. It was a bit tricky to catch when playing with the code.
Maybe you'd see something similar in GL if the target buffer was larger.
There are + and - to capture before or after render. Best would be to capture after any color/transfer conversion and deinterlacing, but before scaling.
There are also pathological videos with incorrect flags / ar that the user works around with video settings overrides. I ignored those in the PR, it makes things very complicated from outside of the renderer code.

Reply:
On GL, it doesn't seem to me that we render to a separate texture, so we are bound to the UI resolution. Maybe this could be problematic when the resolution is set to PAL/NTSC?

If I had the time, I would retool the capture to return the raw YUV frame, and let swscale do the rest. But that's just a remark, and not a problem for this PR.